.github/workflows/build-test.yml

```diff
@@ -333,7 +333,7 @@ jobs:
 
       - name: Build language bindings
         run: |
-          cmake -S . -B target/ffi/build -DBUILD_EXAMPLES=ON -DBUILD_TESTING=ON ${{ matrix.mode.cmake-build-type }} ${{ matrix.mode.cmake-cxx-flags }} ${{ matrix.cmake-build-system-generator }} -DCMAKE_INSTALL_PREFIX=target/ffi/install -DCMAKE_PREFIX_PATH="${{ github.workspace }}/target/iceoryx/install"
+          cmake -S . -B target/ffi/build -DBUILD_EXAMPLES=ON -DBUILD_TESTING=ON -DTREAT_WARNING_AS_ERROR=ON ${{ matrix.mode.cmake-build-type }} ${{ matrix.mode.cmake-cxx-flags }} ${{ matrix.cmake-build-system-generator }} -DCMAKE_INSTALL_PREFIX=target/ffi/install -DCMAKE_PREFIX_PATH="${{ github.workspace }}/target/iceoryx/install"
```

Comment:
Should we stay closer to what we have in iceoryx classic? There the flag is called `BUILD_STRICT`

Reply:
Honestly, I always have to scratch my head and have to figure out `BUILD_STRICT` really means. `TREAT_WARNING_AS_ERROR` is straight forward.

If you don't mind, I would go with it.

Reply:
The idea was to be able to add more constraints than just the warning as error, without having to add more parameter.

What about shortening the parameter to just `WARNING_AS_ERROR`?